consensus/src/round_manager.rs

```diff
@@ -957,26 +946,61 @@ impl RoundManager {
             (Author, Discriminant<VerifiedEvent>),
             (Author, VerifiedEvent),
         >,
+        mut checked_proposal_rx: aptos_channel::Receiver<(), VerifiedEvent>,
         close_rx: oneshot::Receiver<oneshot::Sender<()>>,
     ) {
         info!(epoch = self.epoch_state().epoch, "RoundManager started");
         let mut close_rx = close_rx.into_stream();
         loop {
-            futures::select! {
-                (peer_id, event) = event_rx.select_next_some() => {
-                    let result = match event {
-                        VerifiedEvent::ProposalMsg(proposal_msg) => {
-                            monitor!(
-                                "process_proposal",
-                                self.process_proposal_msg(*proposal_msg).await
-                            )
+            tokio::select! {
+                biased;
```

Comment:
Are you sure this works or should we be using the `select_biased` macro instead?

Reply:
according to this https://michael-f-bryan.github.io/linkcheck/tokio/macro.select.html it should work